skops/card/_model_card.py

```diff
@@ -25,52 +80,79 @@ class Card:
 
+    metadata: CardData, optional
+        ``CardData`` object. The contents of this object are saved as metadata
+        at the beginning of the output file, and used by Hugging Face Hub.
+
+        You can use :func:`~skops.card.metadata_from_config` to create an
+        instance pre-populated with necessary information based on the contents
+        of the ``config.json`` file, which itself is created by
+        :func:`skops.hub_utils.init`.
+
+    Attributes
+    ----------
+    model: estimator object
+        The scikit-learn compatible model that will be documented.
+
+    metadata: CardData
+        Metadata to be stored at the beginning of the saved model card, as
+        metadata to be understood by the Hugging Face Hub.
+
     Notes
     -----
-    You can pass your own custom template using :meth:`Card.add` method. You
-    can add plots to the model card template using :meth:`Card.add_plot`. The
-    key you pass to :meth:`Card.add_plot` will be used for header of the plot.
+    The contents of the sections of the template can be set using
+    :meth:`Card.add` method. Plots can be added to the model card using
+    :meth:`Card.add_plot`. The key you pass to :meth:`Card.add_plot` will be
+    used as the header of the plot.
 
     Examples
     --------
+    >>> import tempfile
+    >>> from pathlib import Path
     >>> from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix
     >>> from sklearn.datasets import load_iris
     >>> from sklearn.linear_model import LogisticRegression
     >>> from skops import card
     >>> X, y = load_iris(return_X_y=True)
     >>> model = LogisticRegression(random_state=0).fit(X, y)
     >>> model_card = card.Card(model)
-    >>> model_card.add(license="mit")  # doctest: +ELLIPSIS
-    <skops.card._model_card.Card object at ...>
+    >>> model_card.metadata.license = "mit"
     >>> y_pred = model.predict(X)
     >>> cm = confusion_matrix(y, y_pred,labels=model.classes_)
-    >>> disp = ConfusionMatrixDisplay(confusion_matrix=cm,
-    ... display_labels=model.classes_)
+    >>> disp = ConfusionMatrixDisplay(
+    ...     confusion_matrix=cm,
+    ...     display_labels=model.classes_
+    ... )
     >>> disp.plot()  # doctest: +ELLIPSIS
     <sklearn.metrics._plot.confusion_matrix.ConfusionMatrixDisplay object at ...>
     >>> disp.figure_.savefig("confusion_matrix.png")
     ...
     >>> model_card.add_plot(confusion_matrix="confusion_matrix.png") # doctest: +ELLIPSIS
     <skops.card._model_card.Card object at ...>
-    >>> model_card.save((Path("save_dir") / "README.md")) # doctest: +ELLIPSIS
+    >>> with tempfile.TemporaryDirectory() as tmpdir:
+    ...     model_card.save((Path(tmpdir) / "README.md")) # doctest: +ELLIPSIS
     ...
     """
 
-    def __init__(self, model: Any, model_diagram: bool = True) -> None:
+    def __init__(
+        self, model: Any, model_diagram: bool = True, metadata: CardData = None
```

Comment:
```suggestion
        self, model: Any, model_diagram: bool = True, metadata: Optional[CardData] = None
```

Probably needs line break...